tracer/src/Datadog.Trace/Propagators/B3MultipleHeaderContextPropagator.cs

```diff
@@ -75,22 +77,22 @@ internal class B3ContextPropagator : IContextInjector, IContextExtractor
             return false;
         }
 
-        private bool IsValidTraceId([NotNullWhen(true)] string? traceId)
+        private static bool IsValidTraceId([NotNullWhen(true)] string? traceId)
         {
             if (string.IsNullOrEmpty(traceId))
             {
                 return false;
             }
 
-            if (traceId!.Length != 16 && traceId!.Length != 32)
+            if (traceId!.Length != 16 && traceId.Length != 32)
```

Comment:
Presumably we don't need the first check either given the IsNullOrEmpty check?
```suggestion
            if (traceId.Length != 16 && traceId.Length != 32)
```

Reply:
The first one is redundant but we need it when targeting .NET Framework to avoid the compiler warnings because `string.IsNullOrEmpty()` is not annotated for nullability.

Rider was just bugging me about the second one, which is doubly redundant. If execution reaches the second condition, then we know for sure `traceId` was not null or accessing `Length` would have thrown a NRE.

Reply:
>The first one is redundant but we need it when targeting .NET Framework to avoid the compiler warnings

le sigh